### DISCUSSION TITLE
apply from-trajectroy-to-model approach to ZnO-water system

### DISCUSSION THREAD

**Discussion by marzieh-saeedimasine** [MarsalekGroup/aml#16] [Q&A]
I used the from-trajectroy-to-model approach for my ZnO- water surface AIMD-DFTB trajectory. I just changed cell and  elements parts in input.nn file and followed all steps (01-03). I applied resulted weights in step 04-C-NNP-MD for MD simulation with METHOD  NNP in CP2K. For some few steps dynamics is good and water molecules are getting to surface and make bound to surface but suddenly temperature increases a lot and makes instability in the system and strange interactions starts to happen.  
I felt that I should exclude some symmetry functions in input.nn file while I train the system. I saw in the example that  it was used triple exclusion exclude_triples = [['O', 'O', 'O']]. Can you say that when we are using this? and for which pair or triple should we use this?
my system is similar to TiO2-Water simulation and I appreciate you if anybody can give me any suggestion about it.

**Comment by cschran** [comment]
Dear Marzieh,

thanks for reaching out again. The `exclude` functionality is only necessary in cases where certain pairs or triples of elements are not present in your system in order to prevent problems with the scaling functionality of n2p2.
For your system I do not expect that exclusion of oxygen triples are required.

In general, you will need to make sure that your training set is representative enough for the envisaged simulations and samples the required time and length scales. In addition, make sure that the training errors are satisfactory.

Best,
Christoph

**Comment by marzieh-saeedimasine** [comment]
Thanks for your answer. Regarding the training set, I increased n_train_initial and n_add to 30 but still the problem is existed. You can see the potential energy of system in AIMD-DFTB in red and predicted MD-MLP in black, for some steps (200 steps) it can predict the energy and dynamics is reasonable but for the rest make explosion in the system. AIMD-DFTB was done for 50000 steps and potential is equilibriated.
 about the training errors, how can I check the training errors? I thought that I can find this information in nnp-train-stdout.err and nnp-scaling-stdout.err files but they are empty. 
Thanks to let me know your idea about it.
![Selection_484](https://user-images.githubusercontent.com/73124374/184651627-199eabb5-fcf3-44ac-b8bc-ffd173fa6a7f.png)

**Comment by cschran** [comment]
the n2p2 train and test errors can be found in `learning-curve.out` for each model `train-00X`.
Typical errors would be ~1 meV/atom for energies and 70 meV/A for forces.

**Reply by marzieh-saeedimasine** [reply]
Thanks for answer. I found  1.91461034E-05 for RMSEpa_Etrain_pu and 6.50282960E-03 for RMSE_Ftrain_pu  for epoch=50 of 03-create-final-model/final-training/train-007 model. I dont know exactly the dimension of RMSEpa_Etrain_pu and RMSE_Ftrain_pu (says physical unit and I guess it would be eV). do you think these values are reasonable?
I realized that I have  Number of biases =41 in my files but in some examples its 51. Can you explain for me how aml.n2p2 determines number of biases?

**Reply by cschran** [reply]
These values should be in atomic units, which you can convert to eV and eV/angstrom to compare to the published values. From my experience, your values in particular for forces seem very high, but please confirm this yourself.

Regarding the biases, if you mean the number of bias nodes, this depends on your NN architecture and descriptors, but this should not make a difference for the final model.
